### ISSUE TITLE
mutex resource leak when freeing DTLS Cookie

### ISSUE THREAD

**Issue by RonEld** [ARMmbed/mbedtls#771]
```
void mbedtls_ssl_cookie_free( mbedtls_ssl_cookie_ctx *ctx )
{
    mbedtls_md_free( &ctx->hmac_ctx );

#if defined(MBEDTLS_THREADING_C)
    mbedtls_mutex_init( &ctx->mutex );
#endif

    mbedtls_zeroize( ctx, sizeof( mbedtls_ssl_cookie_ctx ) );
}
```

`mbedtls_mutex_init` should be change to `mbedtls_mutex_free`
Found by Alan Gillingham in the mbed TLS forum

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTSSL-1162

**Comment by RonEld** [comment]
fixed on versions mbedtls-2.4.2 , mbedtls-2.1.7 and mbedtls-1.3.19
